framework/src/modules/chain/modules/blocks/verify.js

```diff
@@ -835,9 +835,13 @@ Verify.prototype.processBlock = function(block, broadcast, saveBlock, cb) {
 			// 'false' if block comes from chain synchronisation process
 			updateSystemHeaders(seriesCb) {
 				// Update our own headers: broadhash and height
-				!library.config.loading.snapshotRound
-					? components.system.update(seriesCb)
-					: seriesCb();
+				if (!library.config.loading.snapshotRound) {
+					return components.system
+						.update()
+						.then(() => setImmediate(seriesCb))
+						.catch(seriesCb);
```

Comment:
why then and catch has different syntax (one is using then setImmediate, and one not

Reply:
I'm gonna remove `setInmediate`